### PR TITLE
[ZEPPELIN-1921] missing close in closeAndRemoveInterpreterGroup method

### PR DESCRIPTION
**Pull request by cloverhearts**
### What is this PR for?
The problem is that some code in the closeAndRemoveInterpreterGroup method of InterpreterSetting partially closes.

### What type of PR is it?
Bug Fix

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1921

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

**Comment by cloverhearts** [comment]
Hello @jongyoul.

Could you please review for this pr? :)

**Comment by jongyoul** [comment]
@cloverhearts The map of 'interpretergroupRef' couldn't seem to have two value of the same key because it's kind of map. What do you think of it?

**Comment by cloverhearts** [comment]
I was change Linked list to Linked HashSet.

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java [hunk 1]
```diff
@@ -144,6 +144,17 @@ private String getInterpreterProcessKey(String user, String noteId) {
     return key;
   }
 
+  private boolean isEqualInterpreterKey(String noteId, String interpreterGroupKey) {
+    int modeKeyIndex = interpreterGroupKey.lastIndexOf(":");
+    String plainNoteId = interpreterGroupKey.substring(modeKeyIndex, interpreterGroupKey.length());
+
+    if (noteId.equals(plainNoteId)) {
```
Comment (jongyoul) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java]: `return noteId.equals(plainNoteId)` would be fine

**Comment by jongyoul** [comment]
Will merge it if there's no more discussion

**Comment by cloverhearts** [comment]
@jongyoul 
Shows that a part of the test case is forcibly terminated.
I would like to suspend merge until this issue is resolved. :)

**Comment by cloverhearts** [comment]
fixed

**Comment by jongyoul** [comment]
@cloverhearts the failed test looks related this patch. Can you check it?

```
Failed tests: 
  NoteInterpreterLoaderTest.testNoteInterpreterCloseForAll:192 null
```

**Comment by cloverhearts** [comment]
@jongyoul TEST PASS
could you please review for this pr? :)

**Comment by cloverhearts** [comment]
@jongyoul ping~

**Comment by jongyoul** [comment]
Will merge it into master and branch-0.7

**Pull request closed without merging**